icu4j/main/classes/core/src/com/ibm/icu/impl/number/CompactData.java

```diff
@@ -185,8 +185,10 @@ public void put(UResource.Key key, UResource.Value value, boolean isRoot) {
                 // Assumes that the keys are always of the form "10000" where the magnitude is the
                 // length of the key minus one. We expect magnitudes to be less than MAX_DIGITS.
                 byte magnitude = (byte) (key.length() - 1);
+                if (magnitude >= COMPACT_MAX_DIGITS) {
```

Comment:
It looks like this should hopefully "just work". There is a `largestMagnitude` variable, set on line 227 to the largest magnitude present in data. So if you increase `COMPACT_MAX_DIGITS` and other locales don't have the data, they should have a smaller `largestMagnitude` computed at runtime.

Reply:
I filed a separate ticket to make this change: https://unicode-org.atlassian.net/browse/ICU-22143